fastestimator/trace/io/model_eval.py

```diff
@@ -0,0 +1,446 @@
+        self.doc.preamble.append(Command('date', NoEscape(r'\today')))
+        self.doc.append(NoEscape(r'\maketitle'))
+
+        # new column type
+        self.doc.preamble.append(NoEscape(r'\newcolumntype{Y}{>{\centering\arraybackslash}X}'))
+
+        # add tabularx hyphentation
+        self.doc.preamble.append(NoEscape(r'\def\seqinsert{\-}'))
+
+        # TOC
+        self.doc.append(NoEscape(r'\tableofcontents'))
+        self.doc.append(NoEscape(r'\newpage'))
+
+    def _document_test_result(self) -> None:
+        """Document test results including test summary, passed tests, and failed tests.
+        """
+        self.test_id = 1
+        instance_pass_tests, aggregate_pass_tests, instance_fail_tests, aggregate_fail_tests = [], [], [], []
+
+        for test in self.json_summary["tests"]:
+            if test["test_type"] == "per-instance" and test["passed"] == True:
+                instance_pass_tests.append(test)
+            elif test["test_type"] == "per-instance" and test["passed"] == False:
+                instance_fail_tests.append(test)
+            elif test["test_type"] == "aggregate" and test["passed"] == True:
+                aggregate_pass_tests.append(test)
+            elif test["test_type"] == "aggregate" and test["passed"] == False:
+                aggregate_fail_tests.append(test)
+
+        with self.doc.create(Section("Test Summary")):
+            with self.doc.create(Itemize()) as itemize:
+                itemize.add_item(
+                    escape_latex("Execution time: {:.2f} seconds".format(self.json_summary['execution_time(s)'])))
+
+            with self.doc.create(Table(position='H')) as table:
+                table.append(NoEscape(r'\refstepcounter{table}'))
+                self._document_summary_table(pass_num=len(instance_pass_tests) + len(aggregate_pass_tests),
+                                                fail_num=len(instance_fail_tests) + len(aggregate_fail_tests))
+
+        if len(instance_pass_tests) + len(aggregate_pass_tests) > 0:
+            with self.doc.create(Section("Passed Tests")):
+                if len(aggregate_pass_tests) > 0:
+                    with self.doc.create(Subsection("Passed Aggregate Tests")):
+                        with self.doc.create(Table(position='H')) as table:
+                            table.append(NoEscape(r'\refstepcounter{table}'))
+                            self._document_aggregate_table(tests=aggregate_pass_tests)
+                if len(instance_pass_tests) > 0:
+                    with self.doc.create(Subsection("Passed Per-Instance Tests")):
+                        with self.doc.create(Table(position='H')) as table:
+                            table.append(NoEscape(r'\refstepcounter{table}'))
+                            self._document_instance_table(tests=instance_pass_tests, with_ID=self.sample_id)
+
+        if len(instance_fail_tests) + len(aggregate_fail_tests) > 0:
```

Comment:
might want to report failed tests before passed tests

Reply:
got it.

Reply:
fixed